src/main/java/com/softserve/bookstore/repositories/UserRepository.java

```diff
@@ -2,123 +2,148 @@
-                }
-                if (currentUser.getOrders().size() > 1) {
-                    addOrdersToUser(currentUser, users.size());
+                    addRole(Role.USER.toString(), user);
+                    getUserByEmail(users, user.getEmail())
+                            .orElseThrow(() -> new UserNotFoundException("User with email"));
+
+                    User currentUser = getUserByEmail(users, user.getEmail()).get();
+
+                    if (currentUser.getRoles().size() > 1) {
+                        addRole(Role.ADMIN.toString(), user);
+                    }
+                    if (currentUser.getOrders().size() > 1) {
+                        addOrdersToUser(currentUser, users.size());
+                    }
+                } catch (UserNotFoundException | SQLException e) {
+                    throw new RuntimeException(e);
                 }
-            } catch (UserNotFoundException | SQLException e) {
-                throw new RuntimeException(e);
-            }
-        });
+            });
     }
 
 
-    public List<User> findLastUsersAdded(int numberOfRecords) throws SQLException {
-        Connection connection = connectionManager.getConnection();
-        PreparedStatement userStatement = connection.prepareStatement(SELECT_LAST_USERS);
-        userStatement.setInt(1, numberOfRecords);
-        ResultSet resultSet = userStatement.executeQuery();
-        List<User> users = UserUtility.getUsersFromResultSet(resultSet);
-        Collections.reverse(users);
+    private  List<User> getUsersFromResultSet(ResultSet resultSet) throws SQLException {
+        List<User> users = new ArrayList<>();
+        while(resultSet.next()){
+            int userId = resultSet.getInt(User.FIELD_USER_ID);
+            String email = resultSet.getString(User.FIELD_EMAIL).trim();
+            String password = resultSet.getString(User.FIELD_PASSWORD).trim();
+            User user = new User(userId, email, password);
+            users.add(user);
+        }
         return users;
     }
 
+    private void addRole(String roleName, User user) throws SQLException {
+        Connection connection = connectionManager.getConnection();
+
+        Map<String, Integer> roleIds = new HashMap<>();
+        roleIds.put(Role.USER.toString(), 1);
+        roleIds.put(Role.ADMIN.toString(), 2);
+
+        int roleId = Optional.ofNullable(roleIds.get(roleName))
+                .orElseThrow(() -> new IllegalArgumentException("Invalid role name."));
+
+        List<Role> roles = new ArrayList<>();
+        roles.add(Role.valueOf(roleName));
+        user.setRoles(roles);
+
+        PreparedStatement roleStatement = connection.prepareStatement(INSERT_USERS_ROLES);
+        roleStatement.setInt(1, user.getUserId());
+        roleStatement.setInt(2, roleId);
+        roleStatement.executeUpdate();
+        Logger.info("User " + user.getEmail() + " has now role " + user.getRoles());
```

Comment:
consider using placeholders
example -
https://www.baeldung.com/java-logging-tinylog-guide
and also extract constant

Reply:
I modified and I put placeholders in logs where needed.